lib/temporal/workflow/dispatcher.rb

```diff
@@ -13,18 +13,36 @@ class Workflow
     # the event_name. The order of this dispatch is not guaranteed.
     #
     class Dispatcher
+      class DispatchHandler
+        def initialize(handlers_for_target, id)
```

Comment:
nit: It would be handy to add a comment saying that handlers are passed by reference and are mutated here